pkg/apply/task/apply_task.go

```diff
@@ -128,22 +142,48 @@ func (a *ApplyTask) Start(taskContext *taskrunner.TaskContext) {
 	}()
 }
 
-func (a *ApplyTask) sendTaskResult(taskContext *taskrunner.TaskContext, err error) {
-	taskContext.TaskChannel() <- taskrunner.TaskResult{
-		Err: err,
+func newApplyOptions(eventChannel chan event.Event, strategy common.DryRunStrategy, factory util.Factory) (applyOptions, error) {
+	discovery, err := factory.ToDiscoveryClient()
+	if err != nil {
+		return nil, err
+	}
+	dynamic, err := factory.DynamicClient()
+	if err != nil {
+		return nil, err
 	}
-}
 
-func (a *ApplyTask) setApplyOptionsFields(eventChannel chan event.Event) {
-	if ao, ok := a.ApplyOptions.(*apply.ApplyOptions); ok {
-		ao.DryRun = a.DryRunStrategy.ClientDryRun()
-		ao.ServerDryRun = a.DryRunStrategy.ServerDryRun()
-		adapter := &KubectlPrinterAdapter{
+	emptyString := ""
+	return &apply.ApplyOptions{
```

Comment:
It doesn't look like we're setting the `ApplyOptions.Namespace` or `ApplyOptions.EnforceNamespace` here, but I know we set it elsewhere. The `kubectl apply` sets it here using:
```
applyOptions.Namespace, applyOptions.EnforceNamespace, err = factory.ToRawKubeConfigLoader().Namespace()
	if err != nil {
		return nil, err
	}
```
Does it make sense to set it here?

Reply:
I don't think it is needed. Those are used by the Builder and we don't use it (or set it on the ApplyOptions).